util/file.go

```diff
@@ -112,3 +113,33 @@ func OpenTempFile(prefix string, suffix string, mode os.FileMode) (string, *os.F
 	file, err := os.OpenFile(filename, flags, mode)
 	return filename, file, err
 }
+
+// FileExists returns whether the given file or directory exists or not
+func FileExists(path string) (bool, error) {
+	_, err := os.Stat(path)
+	if err == nil {
+		return true, nil
+	}
+	if os.IsNotExist(err) {
+		return false, nil
+	}
+	return false, err
+}
+
+// MakeParentDirs ensures parent directory exist for path
+func MakeParentDirs(path string) error {
+	// 2nd return value here is filename (not an error), which is not needed
+	dir, _ := filepath.Split(path)
+	exists, err := FileExists(dir)
+	if err != nil {
+		return err
+	}
+
+	if !exists {
+		err = os.MkdirAll(dir, 0700)
```

Comment:
Its a directory so needs to be executable.

Added tests for checking file mode.